src/lib.rs

```diff
@@ -748,6 +748,18 @@ impl<A: Array> Drop for IntoIter<A> {
     }
 }
 
+impl<A: Array> Clone for IntoIter<A> where A::Item: Clone {
+    fn clone(&self) -> IntoIter<A> {
+        IntoIter { index: self.index, v: self.v.clone() }
```

Comment:
We have potentially moved out from v, so we can't call clone here

Reply:
Oh, whoops, thank you. Updated to a `.collect().into_iter()`